backend/process_miner/logs_process_miner.py

```diff
@@ -27,12 +30,22 @@
 APPROACH_DEFAULT = "all"
 
 
+def concat_files():
+    """
+        all CSV files get concatenated into one file
+    """
+    all_filenames = glob.glob('retrieved_logs/*.csv')
+    combined_csv = pd.concat([pd.read_csv(f) for f in all_filenames])
+    combined_csv.to_csv("concatenated_files.csv",
+                        index=False, encoding='utf-8-sig')
+
+
 def create_dataframe():
     """
     create dataframe
     """
     dataframe = csv_import_adapter.import_dataframe_from_path(
-        'concated_files.csv', sep=",")
+        'concatenated_files.csv', sep=",")
```

Comment:
You could merge both methods and just directly work with `combined_csv` (and additionally skip all checks for the existence of the merged file)